mapstruct-mapper-repo/generator/src/main/java/org/mapstruct/example/repo/generator/RepositoryGenerator.java

```
@@ -0,0 +1,117 @@
+                    else {
+                        messager.printMessage(
+                                        Diagnostic.Kind.WARNING,
+                                        "@AddToRepo annotated class should implement a StandardMapper",
+                                        annotation
+                        );
+                    }
+                }
+
+            }
+        }
+        if ( roundEnv.processingOver() ) {
+            write( elementUtils.getTypeElement( MapperRepo.class.getName() ) );
+        }
+        return true;
+    }
+
+    private RepoItem toRepoItem(DeclaredType mapper) {
+        RepoItem result = null;
+        for ( Element element : mapper.asElement().getEnclosedElements() ) {
+            if ( element.getKind() == ElementKind.METHOD && "map".equals( element.getSimpleName().toString() ) ) {
+                ExecutableElement method = (ExecutableElement) element;
+                result = new RepoItem( (DeclaredType) method.getParameters().get( 0 ).asType(),
+                                (DeclaredType) method.getReturnType(),
+                                mapper
+                );
+            }
+        }
+        return result;
+    }
+
+    private void write(TypeElement mapperElement) {
+        try (Writer writer = filer.createSourceFile( MapperRepo.class.getName() + "Impl", mapperElement )
+                                  .openWriter()) {
+            Configuration cfg = new Configuration( new Version( "2.3.21" ) );
+            cfg.setClassForTemplateLoading( RepositoryGenerator.class, "/" );
+            cfg.setDefaultEncoding( "UTF-8" );
+
+            Map<String, Object> templateData = new HashMap<>();
+
+            templateData.put( "repoItems", repoItems );
+            Template template = cfg.getTemplate( "MapperRepoImpl.ftl" );
+            template.process( templateData, writer );
+        }
+        catch ( TemplateException | IOException ex ) {
+            ex.printStackTrace();
```

Maybe rethrow the exception?